### PR TITLE
MM-12358 Create CLI command "command modify"

### PR DESCRIPTION
**Pull request by JVasky**
#### Summary

This pull requests add new functionality that allows users to modify an existing command via the Mattermost CLI. This new command is added to improve the functionality of the Mattermost CLI as per ticket MM-12358. This command requires the existing command Id as a parameter and accepts any number of the following flags:

- title
- description
- trigger-word
- url
- creator
- response-username
- icon
- autocomplete 
- post

Something to consider with this PR is that if the flags --autocomplete and --post are not specified, the specified command will be updated to false and "G" respectively, as per their function via the "command create" command.  This includes modify commands that have no flags assigned.

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/9597

### CODE REVIEW

**Comment by hanzei** [comment]
@JVasky Is this PR ready for review? Do you have outstanding questions?

**Comment by JVasky** [comment]
@hanzei I don't have any outstanding questions at this time so I think this PR is ready for review!

File: cmd/mattermost/commands/command.go [hunk 1]
```diff
@@ -268,3 +290,97 @@ func deleteCommandCmdF(command *cobra.Command, args []string) error {
...
+	// get creator user
+	creator, _ := command.Flags().GetString("creator")
+	if creator != "" {
+		user := getUserFromUserArg(a, creator)
+		if user == nil {
+			return errors.New("unable to find user '" + creator + "'")
+		}
+
+		// check if creator has permission to create slash commands
+		if !a.HasPermissionToTeam(user.Id, modifiedCommand.TeamId, model.PERMISSION_MANAGE_SLASH_COMMANDS) {
```
Comment (jespino) [inline cmd/mattermost/commands/command.go]: I'm ok with this for this PR because is already checked in the `CreateCommandCmd` but we normally avoid permissions checks and things like that in the CLI command, because if you have CLI command we understand that you have full privileges. Anyway, as I said, this should be discussed and fixed in another PR, for now I prefer the existing consistency between both commands.
